File: e-nutrition-symfony/src/Entity/FicheConsultation.php
<?php

namespace App\Entity;

use App\Repository\FicheConsultationRepository;
use App\Repository\PatientRepository;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use phpDocumentor\Reflection\Types\Array_;

/**
 * @ORM\Entity(repositoryClass=FicheConsultationRepository::class)
 */
class FicheConsultation
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="date")
     */
    private $CreationDate;

    /**
     * @ORM\Column(type="float")
     */
    private $Poids;

    /**
     * @ORM\Column(type="float")
     */
    private $Taille;



    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $Symptome;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Apetit;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $Description;

    /**
     * @ORM\OneToMany(targetEntity=Medicament::class, mappedBy="fiche",cascade={"all"},orphanRemoval=true)
     */
    private $medicaments;

    /**
     * @ORM\OneToMany(targetEntity=TagFicheConsultation::class, mappedBy="ficheConsultation",cascade={"all"},orphanRemoval=true)
     */
    private $tagFicheConsultation;

    /**
     * @ORM\ManyToOne(targetEntity=Patient::class, inversedBy="ficheConsultations")
     */
    private $patient;

    /**
     * @ORM\ManyToOne(targetEntity=Nutritionniste::class, inversedBy="ficheConsultations")
     */
    private $nutritionniste;

    public function __construct()
    {
        $this->medicaments = new ArrayCollection();
        $this->tagFicheConsultation = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCreationDate(): ?\DateTimeInterface
    {
        return $this->CreationDate;
    }

    public function setCreationDate(\DateTimeInterface $CreationDate): self
    {
        $this->CreationDate = $CreationDate;

        return $this;
    }

    public function getPoids(): ?float
    {
        return $this->Poids;
    }

    public function setPoids(float $Poids): self
    {
        $this->Poids = $Poids;

        return $this;
    }

    public function getTaille(): ?float
    {
        return $this->Taille;
    }

    public function setTaille(float $Taille): self
    {
        $this->Taille = $Taille;

        return $this;
    }


    public function getSymptome(): ?string
    {
        return $this->Symptome;
    }

    public function setSymptome(?string $Symptome): self
    {
        $this->Symptome = $Symptome;

        return $this;
    }

    public function getApetit(): ?string
    {
        return $this->Apetit;
    }

    public function setApetit(string $Apetit): self
    {
        $this->Apetit = $Apetit;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(?string $Description): self
    {
        $this->Description = $Description;

        return $this;
    }

    /**
     * @return Collection|Medicament[]
     */
    public function getMedicaments(): Collection
    {
        return $this->medicaments;
    }

    public function addMedicament(Medicament $medicament): self
    {
        if (!$this->medicaments->contains($medicament)) {
            $this->medicaments[] = $medicament;
            $medicament->setFiche($this);
        }

        return $this;
    }

    public function removeMedicament(Medicament $medicament): self
    {
        if ($this->medicaments->removeElement($medicament)) {
            // set the owning side to null (unless already changed)
            if ($medicament->getFiche() === $this) {
                $medicament->setFiche(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection|TagFicheConsultation[]
     */
    public function getTagFicheConsultation(): Collection
    {
        return $this->tagFicheConsultation;
    }

    public function addTagFicheConsultation(TagFicheConsultation $tagFicheConsultation): self
    {
        if (!$this->tagFicheConsultation->contains($tagFicheConsultation)) {
            $this->tagFicheConsultation[] = $tagFicheConsultation;
            $tagFicheConsultation->setFicheConsultation($this);
        }

        return $this;
    }

    public function removeTagFicheConsultation(TagFicheConsultation $tagFicheConsultation): self
    {
        if ($this->tagFicheConsultation->removeElement($tagFicheConsultation)) {
            // set the owning side to null (unless already changed)
            if ($tagFicheConsultation->getFicheConsultation() === $this) {
                $tagFicheConsultation->setFicheConsultation(null);
            }
        }

        return $this;
    }

    public function getPatient(): ?Patient
    {
        return $this->patient;
    }

    public function setPatient(?Patient $patient): self
    {
        $this->patient = $patient;

        return $this;
    }

    public function getNutritionniste(): ?Nutritionniste
    {
        return $this->nutritionniste;
    }

    public function setNutritionniste(?Nutritionniste $nutritionniste): self
    {
        $this->nutritionniste = $nutritionniste;

        return $this;
    }


    public function calculerImc() :float
    {
        return         number_format(($this->getTaille())/($this->getPoids()*$this->getPoids()),2);
    }

    public function  getnbrOFTimeUsed():float
    {
        $j=0;
        foreach ($this->getPatient() ->getFicheConsultations() as $ficheConsultation )
        {
            $j++;
        }
        return$j;
    }



}

File: e-nutrition-java/src/e/nutrition/Models/FicheConsultation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package e.nutrition.Models;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Admin
 */
public class FicheConsultation {
    private int id;
    private Date creation_date;
    private float poids;
    private float taille;
    private String symptome;
    private String apetit;
      private String description;
 
   private List <Medicament> medicaments = new ArrayList();

    public FicheConsultation(int id, Date creation_date, float poids, float taille, String symptome, String apetit, String description) {
        this.id = id;
        this.creation_date = creation_date;
        this.poids = poids;
        this.taille = taille;
        this.symptome = symptome;
        this.apetit = apetit;
        this.description = description;
    }

    public FicheConsultation(Date creation_date, float poids, float taille, String symptome, String apetit, String description) {
        this.creation_date = creation_date;
        this.poids = poids;
        this.taille = taille;
        this.symptome = symptome;
        this.apetit = apetit;
        this.description = description;
    }
    
    
    public List<Medicament> getMedicaments() {
        return medicaments;
    }

    public void setMedicaments(List<Medicament> medicaments) {
        this.medicaments = medicaments;
    }
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getCreation_date() {
        return creation_date;
    }

    public void setCreation_date(Date creation_date) {
        this.creation_date = creation_date;
    }

    public float getPoids() {
        return poids;
    }

    public void setPoids(float poids) {
        this.poids = poids;
    }

    public float getTaille() {
        return taille;
    }

    public void setTaille(float taille) {
        this.taille = taille;
    }

    public String getSymptome() {
        return symptome;
    }

    public void setSymptome(String symptome) {
        this.symptome = symptome;
    }

    public String getApetit() {
        return apetit;
    }

    public void setApetit(String apetit) {
        this.apetit = apetit;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "FicheConsultation{" + "id=" + id + ", creation_date=" + creation_date + ", poids=" + poids + ", taille=" + taille + ", symptome=" + symptome + ", apetit=" + apetit + ", description=" + description + '}';
    }

    
   
   
    
    
    
    
    
    
    
    
    
}
